app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -147,7 +147,14 @@ def pii_requested_but_locked?
     end
 
     def track_events
-      analytics.track_event(Analytics::SP_REDIRECT_INITIATED, ial: sp_session_ial)
+      event_ial_context = @authorize_form.ial_context.dup
```

Comment:
It seems a little messy to have to dupe and modify the ial context here. Would it perhaps make more sense to make the form aware of the user and pass it in there?

Reply:
It might just be easier to make a new `ial_context`, reference the old one for its parsing?
```suggestion
      event_ial_context = IalContext.new(
        service_provider: @authorize_form.service_provider,
        user: current_user,
        ial: @authorize_form.ial, 
      )
```